test/export/test_torchbind.py

```diff
@@ -1,36 +1,83 @@
+from torch._functorch.aot_autograd import aot_export_module
+from torch._higher_order_ops.effects import _EffectType, SIDE_EFFECTS
 from torch._higher_order_ops.torchbind import enable_torchbind_tracing
 from torch.export import export
-from torch.testing._internal.common_utils import (
-    find_library_location,
-    IS_FBCODE,
-    IS_MACOS,
-    IS_SANDCASTLE,
-    IS_WINDOWS,
-    run_tests,
-    skipIfTorchDynamo,
-    TestCase,
-)
+from torch.fx.experimental.proxy_tensor import make_fx
+from torch.testing._internal.common_utils import run_tests, skipIfTorchDynamo, TestCase
 
 
 @skipIfTorchDynamo("torchbind not supported with dynamo yet")
 class TestExportTorchbind(TestCase):
     def setUp(self):
-        if IS_MACOS:
-            raise unittest.SkipTest("non-portable load_library call used in test")
-        elif IS_SANDCASTLE or IS_FBCODE:
-            torch.ops.load_library(
-                "//caffe2/test/cpp/jit:test_custom_class_registrations"
-            )
-        elif IS_WINDOWS:
-            lib_file_path = find_library_location("torchbind_test.dll")
-            torch.ops.load_library(str(lib_file_path))
-        else:
-            lib_file_path = find_library_location("libtorchbind_test.so")
-            torch.ops.load_library(str(lib_file_path))
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_Foo")
+        class FakeFoo:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+            @classmethod
+            def from_real(cls, foo):
+                (x, y), _ = foo.__getstate__()
+                return cls(x, y)
+
+            def add_tensor(self, z):
+                return (self.x + self.y) * z
+
+        @torch._library.impl_abstract_class("_TorchScriptTesting::_TensorQueue")
+        class FakeTensorQueue:
+            def __init__(self, q):
+                self.queue = q
+
+            @classmethod
+            def from_real(cls, real_tq):
+                return cls(real_tq.clone_queue())
+
+            def push(self, x):
+                self.queue.append(x)
+
+            def pop(self):
+                return self.queue.pop(0)
+
+            def size(self):
+                return len(self.queue)
+
+    def tearDown(self):
+        torch._library.abstract_impl_class.deregister_abstract_impl(
+            "_TorchScriptTesting::_Foo"
+        )
+        torch._library.abstract_impl_class.deregister_abstract_impl(
+            "_TorchScriptTesting::_TensorQueue"
+        )
+
+    def _assert_equal_skip_script_object(self, exp, actual):
+        flat_exp = pytree.tree_leaves(exp)
+        flat_actual = pytree.tree_leaves(actual)
+        self.assertEqual(len(flat_exp), len(flat_actual))
+        for a, b in zip(flat_exp, flat_actual):
+            if isinstance(a, torch.ScriptObject) and isinstance(b, torch.ScriptObject):
+                continue
+            self.assertEqual(a, b)
+
+    def _test_export_same_as_eager_skip_script_object(
```

Comment:
We have a lot of ways of naming the same thing. We should try to be consistent when we're creating new APIs. I've seen "script object", "custom class", "torchbind class", etc...

We should probably just call it TorchBind{Object/Class}

Reply:
the class name is torch.ScriptObject though..hahah But never mind. TorchBindObject/Class seems good to me.

Reply:
So we do have the option of changing torch.ScriptObject into torch.TorchBindObject. But torch.TorchBindObject sounds a bit wild... should we call it "CustomClass/CustomObject" then?